guava/src/main/java/com/fasterxml/jackson/datatype/guava/GuavaDeserializers.java

```diff
@@ -258,9 +261,6 @@ public JsonDeserializer<?> findMapLikeDeserializer(MapLikeType type,
                     elementTypeDeserializer, elementDeserializer);
         }
 
-        if (Table.class.isAssignableFrom(raw)) {
```

Comment:
@cowtowncoder, should we use MapLikeType/MapLikeDeserializer or not? What I found is if we keep it MapLikeType, then we have to use Type something like `Map<R, Map<C, V>>`, and then we'll have KeyDeserializer for RowKey and JsonDeserializer for ColumnValueMap, and then while deserializing the `Table<R, C, V>` we'll have to deserialize `Map<C, V>`  first and then have to put each `C, V` in `Table<R, C, V>` for `R`, which is inefficient in my opinion. Contrarily, if we don't use MapLikeType and keep it JavaType (Using Bean Deserializer), we can deserialize directly to `Table<R, C, V>`. 

What do you think about this?

Reply:
Hmmmh. Tough question. I think that:

1. Type should be retained as `MapLikeType` (to help Jackson databind be somewhat aware of type's semantic)
2. But! We do NOT need to use `MapLikeDeserializer`, if that is possible

If (2) can't be done for some reason (I may be overlooking something), I'm ok in change to make Tables not be considered "Map-like". This will add some work on (de)serialization handling, possibly.

Reply:
Sure, I have used `MapLikeType` without using any other MapLikeType deserialization arguments, as if we use elementDeserializer, then we might have to first deserialize to `Map<C, V>` and then have to put in `Table<R,C,V>`, since we cannot get `C`'s deserializer and `V`'s deserializer separately from `JsonDeserializer<?> elementDeserializer` of this API:

```     
public JsonDeserializer<?> findMapLikeDeserializer(MapLikeType type, DeserializationConfig config, BeanDescription beanDesc, KeyDeserializer keyDeserializer, TypeDeserializer elementTypeDeserializer, JsonDeserializer<?> elementDeserializer)
```

Well, this is from my limited knowledge of this codebase; please do correct me if I'm missing something here.

Reply:
That sounds good -- hoping to review this in near future, just having bit of overload with everything going on. But thank you again for this contribution!

Reply:
Thanks for the update! I completely understand, and no worries at all. Take your time. I’m happy to contribute and look forward to any feedback whenever you’re able to review. Thanks again!